scss/_mixins.scss

```diff
@@ -0,0 +1,269 @@
+	}
+
+	.o-topper--basic {
+		@include oTopperBasic;
+	}
+
+	.o-topper__headline {
+		@include oTopperHeadline;
+	}
+
+	.o-topper__headline--large {
+		@include oTopperHeadlineLarge;
+	}
+
+	.o-topper__summary {
+		@include oTopperSummary;
+	}
+
+	.o-topper__standfirst {
+		@include oTopperStandfirst;
+	}
+
+	.o-topper__summary--body {
+		@include oTopperSummaryBody;
+	}
+
+	.o-topper__tags {
+		@include oTopperTags;
+	}
+
+	.o-topper__columnist {
+		@include oTopperColumnist;
+	}
+
+	.o-topper__columnist-name {
+		@include oTopperColumnistName;
+	}
+
+	.o-topper__brand {
+		@include oTopperBrand;
+	}
+
+	.o-topper__topic {
+		@include oTopperTopic;
+	}
+
+	.o-topper__read-next {
+		@include oTopperReadNext;
+	}
+
+	.o-topper__image {
+		@include oTopperImage;
+	}
+
+	.o-topper__image-credit {
+		@include oTopperImageCredit;
+	}
+
+	.o-topper__anchor-link {
+		@include oTopperAnchorLink;
+	}
+}
+
+@mixin oTopperColors() {
+	@each $color in $_o-topper-colors {
+		.o-topper--color-#{$color} {
+			@include oTopperColor($color);
+		}
+	}
+}
+
+// uses topper background colour to set background and pick contrasting text color
+@mixin oTopperColor($color-name) {
+	$background: oColorsGetPaletteColor($color-name);
+	$foreground: oColorsGetTextColor($background, $opacity: 100);
+	$is-light: $foreground == 'white';
+	$hover-color: lighten($foreground, 40%);
+
+	@if $is-light {
+		$hover-color: darken($foreground, 20%);
+	}
+
+	// include oTopperColor on a parent then extend this placeholder on a
+	// descendent to lend that descendent the color you specified in the mixin
+	.o-topper__background,
+	.o-topper__content {
+		@if $color-name == 'wheat' {
+			background-color: oColorsMix($color: 'wheat', $percentage: 40);
+		} @else {
+			background-color: $background;
+		}
+	}
+
+	@if not $is-light {
+		.o-topper__standfirst {
+			a {
+				color: $foreground;
+				border-bottom-color: $foreground;
+
+				&:hover {
+					color: $hover-color;
+					border-bottom-color: $hover-color;
+				}
+			}
+		}
+	}
+
+	.o-topper__headline {
+		&::after {
+			border-bottom: 6px solid $foreground;
+
+			@include oGridRespondTo($from: L) {
+				border-bottom-width: 8px;
+			}
+		}
+	}
+
+	.o-topper__tags,
+	.o-topper__columnist-name,
+	.o-topper__headline,
+	.o-topper__standfirst,
+	.o-topper__summary {
+		color: $foreground;
+	}
+
+	.o-topper__topic,
+	.o-topper__brand {
+		color: $foreground;
+
+		&:hover {
+			color: oColorsMix($foreground, $color-name, 73);
+		}
+	}
+
+	@if $color-name == 'paper' or $color-name == 'wheat' {
+		.o-topper__topic,
+		.o-topper__columnist-name {
+			@include oColorsFor('tag-link', text);
+
+			&:hover {
+				@include oColorsFor('tag-link-hover', text);
+			}
+		}
+	}
+
+	@if $color-name == 'sky' {
+		.o-topper__tags,
+		.o-topper__topic,
+		.o-topper__columnist-name {
+			color: oColorsGetPaletteColor('oxford');
+
+			&:hover {
+				color: oColorsGetPaletteColor('black-80');
+			}
+		}
+	}
+}
+
+@mixin _oTopperAlignText($position) {
```

Comment:
Minor suggestion: `mixins.scss` holds the high level "output all the things" mixins. `_oTopperAlignText`, `_oTopperDefineOnce` feel a bit out of place. Put them closer to where they're used or in a utils file?

Reply:
personally i find it confusing when a component defines its mixins scattered around a bunch of files? makes it harder to tell what's available at a glance